module/zfs/dmu_tx.c

```diff
@@ -43,17 +43,18 @@ typedef void (*dmu_tx_hold_func_t)(dmu_tx_t *tx, struct dnode *dn,
     uint64_t arg1, uint64_t arg2);
 
 dmu_tx_stats_t dmu_tx_stats = {
-	{ "dmu_tx_assigned",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_delay",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_error",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_suspended",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_group",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_memory_reserve",	KSTAT_DATA_UINT64 },
-	{ "dmu_tx_memory_reclaim",	KSTAT_DATA_UINT64 },
-	{ "dmu_tx_dirty_throttle",	KSTAT_DATA_UINT64 },
-	{ "dmu_tx_dirty_delay",		KSTAT_DATA_UINT64 },
-	{ "dmu_tx_dirty_over_max",	KSTAT_DATA_UINT64 },
-	{ "dmu_tx_quota",		KSTAT_DATA_UINT64 },
+	{ "dmu_tx_assigned",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_delay",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_error",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_suspended",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_group",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_memory_reserve",		KSTAT_DATA_UINT64 },
+	{ "dmu_tx_memory_reclaim",		KSTAT_DATA_UINT64 },
+	{ "dmu_tx_dirty_throttle",		KSTAT_DATA_UINT64 },
+	{ "dmu_tx_dirty_delay",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_dirty_over_max",		KSTAT_DATA_UINT64 },
+	{ "dmu_tx_quota",			KSTAT_DATA_UINT64 },
+	{ "dmu_tx_per_txg_dirty_frees_delay",	KSTAT_DATA_UINT64 },
```

Comment:
Good idea, this would be really helpful for tuning.  I'd just recommend dropping the `_per_txg` portion of the name which is a bit misleading since these are global counters for the system.

Reply:
agree, otherwise LGTM